src/models/endpoint.js

```diff
@@ -206,6 +217,20 @@ function compareHashMaps(configured, incoming) {
   return headers;
 }
 
+function compareObjects(configured, incoming) {
+  var key;
+
+  for (key in configured) {
+    if (typeof configured[key] !== typeof incoming[key]) { return false; }
```

Comment:
I'm admittedly being a bit lazy with this function, but it's lazy with a reason.  Valid JSON cannot actually parse to NaN or Infinity based on the spec (http://tools.ietf.org/html/rfc4627 Section 2.4).